src/Func.h

```diff
@@ -279,7 +281,15 @@ class Stage {
     EXPORT Stage &allow_race_conditions();
 
     EXPORT Stage &hexagon(VarOrRVar x = Var::outermost());
-    EXPORT Stage &prefetch(VarOrRVar var, Expr offset = 1);
+    EXPORT Stage &prefetch(const Func &f, VarOrRVar var, Expr offset = 1,
+                           PrefetchBoundStrategy strategy = PrefetchBoundStrategy::GuardWithIf);
+    EXPORT Stage &prefetch(const Internal::Parameter &param, VarOrRVar var, Expr offset = 1,
+                           PrefetchBoundStrategy strategy = PrefetchBoundStrategy::GuardWithIf);
+    template<typename T>
+    Stage &prefetch(const T &image, VarOrRVar var, Expr offset = 1,
```

Comment:
Should this parameter be Input<Buffer<T>> to avoid strange errors (parameter is not a method on some arbitrary thing passed in)?

@steven-johnson ?